cumulus_etl/etl/tasks/nlp_task.py

```diff
@@ -0,0 +1,83 @@
+"""Base NLP task support"""
+
+import copy
+import logging
+import os
+from typing import Callable
+
+import rich.progress
+
+from cumulus_etl import common, fhir, nlp, store
+from cumulus_etl.etl.tasks.base import EtlTask, OutputTable
+
+
+class BaseNlpTask(EtlTask):
+    """Base class for any clinical-notes-based NLP task."""
+
+    resource = "DocumentReference"
+    needs_bulk_deid = False
+
+    # You may want to override these in your subclass
+    outputs = [OutputTable(schema=None)]  # maybe a group_field? (remember to call self.seen_docrefs.add() if so)
+    tags = {"gpu"}  # maybe a study identifier?
+
+    # Task Version
+    # The "task_version" field is a simple integer that gets incremented any time an NLP-relevant parameter is changed.
+    # This is a reference to a bundle of metadata (model revision, container revision, prompt string).
+    # We could combine all that info into a field we save with the results. But it's more human-friendly to have a
+    # simple version to refer to.
+    #
+    # CONSIDERATIONS WHEN CHANGING THIS:
+    # - Record the new bundle of metadata in your class documentation
+    # - Update any safety checks in prepare_task() or elsewhere that check the NLP versioning
+    # - Be aware that your caching will be reset
+    task_version = 1
```

Comment:
if you want this to start at 0 (like in hf_tasks), this should also start as 0?

Reply:
I was mentally treating 0 vs 1 like semantic versioning does - that at 1 you are opting into actually caring about breaks but at zero you don't need to care about them.

So I was just thinking I'd want to push subclasses to be well-versioned from the start (hf_test is a toy thing that will never be a real boy, so it doesn't matter).